src/cgr_gwas_qc/workflow/scripts/vcf2abf.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+import typer
+from pysam import VariantFile
+
+app = typer.Typer(add_completion=False)
+
+
+# In[2]:
+
+
+def is_biallelic_snp(rec):
+    return len(rec.ref) == 1 and len(rec.alts[0]) == 1
+
+
+# In[27]:
+
+
+@app.command()
+def main(
+    bcf_file: Path = typer.Argument(
+        ..., help="Path to the sample VCF.", exists=True, readable=True
+    ),
+    kgvcf_file: Path = typer.Argument(
+        ..., help="Path to the 1000 genomes VCF file to get ABF.", exists=True, readable=True
+    ),
+    population: str = typer.Argument(
+        ...,
+        help="Which population in VCF file to use for ABF [Not implemented,global/AF used currently].",
+    ),
+    abf_file: Optional[Path] = typer.Argument(
+        None, help="Output file name. If none given it will use the `VCF_FILE + .abf.txt`."
+    ),
+):
+    """Pull out the B allele frequency for a given subpopulation.
+
+    Takes variants from the provided `vcf_file` and tries to find their B
+    allele frequency in the provided `kgvcf_file`. The B allele frequencies are
+    based on the allele frequency for the provided subpopulation
+    `population`. If the provided subpopulation doesn't exist, it will output
+    a list of available populations. If the variant does not exists in the
+    `vcf_file` then the ABF will be set to "NA".
+    """
+
+    if abf_file is None:
+        abf_file = bcf_file.with_suffix(".abf.txt")
+
+    # vcf_file='/scratch/rajwanir2/data_catalog/QC-test/results/GSAMD-24v1-0-contamseries/vcf/GSAMD-24v1-0-contamseries.bcf'
```

Comment:
Remove these debugging lines.

Reply:
Removed in 874eb15e1e0fac826a01450e7ed8898c63cc112d. Thanks.